x-pack/platform/plugins/shared/streams/server/plugin.ts

```diff
@@ -72,6 +76,37 @@ export class StreamsPlugin
     const assetService = new AssetService(core, this.logger);
     const streamsService = new StreamsService(core, this.logger, this.isDev);
 
+    plugins.features.registerKibanaFeature({
+      id: 'streams',
+      name: i18n.translate('xpack.streams.featureRegistry.streamsFeatureName', {
+        defaultMessage: 'Streams',
+      }),
+      order: 600,
+      category: DEFAULT_APP_CATEGORIES.observability,
+      scope: [KibanaFeatureScope.Spaces, KibanaFeatureScope.Security],
+      app: ['streams'],
+      privileges: {
+        all: {
+          app: ['streams'],
+          savedObject: {
+            all: [],
+            read: [],
+          },
+          api: [STREAMS_FEATURE_PRIVILEGES.assets],
+          ui: [STREAMS_FEATURE_PRIVILEGES.show, STREAMS_FEATURE_PRIVILEGES.assets],
+        },
+        read: {
+          app: ['streams'],
+          savedObject: {
+            all: [],
+            read: [],
+          },
+          api: [],
+          ui: [STREAMS_FEATURE_PRIVILEGES.show],
```

Comment:
Does it mean we never enforce privileges on read APIs, and instead delegate to es? And for "write" api, we do check for the privilege?

Reply:
Wondering if we should have base `read` privilege on all apis not already gated ? There's something off in not being able to read the UI but still being able to call the read apis

Reply:
> And for "write" api, we do check for the privilege?

Only for the ones that change the dashboard links.

> Wondering if we should have base read privilege on all apis not already gated ? There's something off in not being able to read from the UI but still being able to call the read apis

It's not adding anything they couldn't do directly talking to Elasticsearch, but I think you are right - it's weird that these other APIs are still available. I'm going to add that.